lib/ex_aws/sqs/client.ex

```diff
@@ -0,0 +1,158 @@
+
+  @doc "Purge all messages in a SQS Queue"
+  defcallback purge_queue(queue_name :: binary) :: ExAws.Request.response_t
+
+  @doc """
+  Adds a permission with the provided label to the Queue
+  for a specific action for a specific account.
+  """
+  defcallback add_permission(queue_name :: binary, label :: binary, permissions :: sqs_acl) :: ExAws.Request.response_t
+
+  @doc "Removes permission with the given label from the Queue"
+  defcallback remove_permission(queue_name :: binary, label :: binary) :: ExAws.Request.response_t
+
+  @doc "Send a message to a SQS Queue"
+  @type sqs_message_opts :: [
+      {:delay_seconds, 0..900} |
+      {:attributes, sqs_message_attribute | [sqs_message_attribute, ...]}
+  ]
+  defcallback send_message(queue_name :: binary, message_body :: binary) :: ExAws.Request.response_t
+  defcallback send_message(queue_name :: binary, message_body :: binary, opts :: sqs_message_opts) :: ExAws.Request.response_t
+
+  @doc "Send up to 10 messages to a SQS Queue in a single request"
+  @type sqs_batch_message :: binary |
+  [
+    {:message_body, binary} |
+    {:delay_seconds, 0..900} |
+    {:attributes, sqs_message_attribute | [sqs_message_attribute, ...]}
+  ]
+  defcallback send_message_batch(queue_name :: binary, messages :: [sqs_batch_message, ...]) :: ExAws.Request.response_t
+
+  @doc "Read messages from a SQS Queue"
+  @type receive_message_opts :: [
+    {:attribute_names, sqs_message_attribute_name | [sqs_message_attribute_name, ...]} |
+    {:max_number_of_messages, 1..10} |
+    {:visibility_timeout, 0..43200} |
+    {:wait_timeout, 0..20}
+  ]
+  defcallback receive_message(queue_name :: binary) :: ExAws.Request.response_t
+  defcallback receive_message(queue_name :: binary, opts :: receive_message_opts) :: ExAws.Request.response_t
+
+  @doc "Delete a message from a SQS Queue"
+  defcallback delete_message(queue_name :: binary, receipt_handle :: binary) :: ExAws.Request.response_t
+
+  @doc "Deletes a list of messages from a SQS Queue in a single request"
+  defcallback delete_message_batch(queue_name :: binary, receipt_handles :: [binary, ...]) :: ExAws.Request.response_t
+
+  @doc """
+  Extends the read lock timeout for the specified message from
+  the specified queue to the specified value
+  """
+  defcallback change_message_visibility(queue_name :: binary, receipt_handle :: binary, visibility_timeout :: visibility_timeout) :: ExAws.Request.response_t
+
+  @doc """
+  Extends the read lock timeout for a batch of 1..10 messages
+  """
+  @type message_visibility_batch :: %{
+    :id => binary,
+    :receipt_handle => binary,
+    :visibility_timeout => visibility_timeout
+  }
+  defcallback change_message_visibility_batch(queue_name :: binary, opts :: [message_visibility_batch, ...]) :: ExAws.Request.response_t
+
+  defmacro __using__(opts) do
+    boilerplate = __MODULE__
+    |> ExAws.Client.generate_boilerplate(opts)
+
+    quote do
```

Comment:
Per changes upcoming in v0.2.0 this should have a line with `defstruct config: nil, service: :sqs` prior to unquoting the boilerplate. Because of this, the `service/0` can be deleted.